### PR TITLE
[telemetry] add protobuf for upstream DNS metrics

### PR DESCRIPTION
**Pull request by sherysheng**
Add the counters of queries, responses, failures handled by upstream DNS server

### CODE REVIEW

File: src/dbus/server/dbus_thread_object.cpp [hunk 1]
```diff
@@ -1394,6 +1394,10 @@ otError DBusThreadObject::GetDnssdCountersHandler(DBusMessageIter &aIter)
 
     dnssdCounters.mResolvedBySrp = otDnssdCounters.mResolvedBySrp;
```
Comment (morningboata) [inline src/dbus/server/dbus_thread_object.cpp:1396]: If you want to change the DBus interface, you need to modify the DBus interface definition at src/dbus/server/introspect.xml, DnssdCounters type. Double check if the client can talk correctly.
If you don't want to change the DBus interface, avoid the content serialization changes.

Comment (sherysheng) [reply src/dbus/server/dbus_thread_object.cpp:1396]: I need to add three more metrics here, seems that I have to change the struct ?

    <!-- DnssdCounters: The DNS-SD counters
    <literallayout>
        struct {
          uint32 success
          uint32 server_failure
          uint32 format_error
          uint32 name_error
          uint32 not_implemented
          uint32 other
          uint32 resolved_by_srp

Comment (erjiaqing) [reply src/dbus/server/dbus_thread_object.cpp:1396]: I would suggest not do to so, or in separate PRs.

DBus clients will not work if this is changed unless the users updated their code.

Comment (sherysheng) [reply src/dbus/server/dbus_thread_object.cpp]: Thanks! Got it

**Review by morningboata — APPROVED** [review]

**Comment by sherysheng** [comment]
Depends-on https://github.com/openthread/openthread/pull/9729

**Review by erjiaqing — APPROVED** [review]
Thanks!

**Review by Irving-cl — APPROVED** [review]
LGTM 👍

**Review by abtink — APPROVED** [review]
👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**